### DISCUSSION TITLE
Documentation of/ insight into ens protocol

### DISCUSSION THREAD

**Discussion by thomaskilian** [john30/ebusd#1318] [Q&A] [answered]
With a little experimenting I found that the enh protocoll is a simple socket communication that echoes the ebus traffic and vice versa sends bytes onto the bus. The ens protocol (which seems to be required by ebusd) is just  a conundrum. Someone here how has some pointers?

**Comment by john30** [comment, marked as answer]
see here https://github.com/john30/ebusd/wiki/2.-Run#device-options
and here https://github.com/john30/ebusd/blob/master/docs/enhanced_proto.md

**Reply by john30** [reply]
and it is not required as can be seen in the command line help as well as the run option docs, looking at the documentation always helps btw

**Reply by thomaskilian** [reply]
Well, it was by experience. I tried to use ebusd with enh but it complained. I might have been doing something wrong with the start, though. I'll try that again...

**Reply by thomaskilian** [reply]
Just retried and it worked. I might have accidentally prepended enh. Now with the plain ip it worked. The ebusd add-in on HA suggested that enh prefix, so that's probably the culprit.
P.S. opened an issue over there.
